crates/wasmedge/src/instance.rs

```diff
@@ -186,6 +207,70 @@ impl Instance for Wasi {
         }
     }
     fn start(&self) -> Result<u32, Error> {
+        // Call prehook before the start
+        let bundle = self.bundle.clone();
+        let spec = oci::load(Path::new(&bundle).join("config.json").to_str().unwrap())?;
+        match spec.hooks().as_ref() {
```

Comment:
I feel like hook calls should maybe go up a layer so that all shims will get it?

Reply:
Done with 21a96e075315df1b4e73a33eaed6b106f0796b2d.